OpenRA.Mods.Common/Widgets/Logic/Ingame/CommandBarLogic.cs

```diff
@@ -40,10 +42,14 @@ public class CommandBarLogic : ChromeLogic
 		TraitPair<IIssueDeployOrder>[] selectedDeploys = { };
 
 		[ObjectCreator.UseCtor]
-		public CommandBarLogic(Widget widget, World world)
+		public CommandBarLogic(Widget widget, World world, Dictionary<string, MiniYaml> logicArgs)
 		{
 			this.world = world;
 
+			var highlightOnButtonPress = false;
+			if (logicArgs.ContainsKey("HighlightOnButtonPress"))
+				highlightOnButtonPress = FieldLoader.GetValue<bool>("HighlightOnButtonPress", logicArgs["HighlightOnButtonPress"].Value);
```

Comment:
I'm not sure about this: in principle I agree that it would be cleaner, but in practice we only use it in this one place, and it would be inconsistent with other uses of the LogicArgs (e.g. #13711 where the key names also need to be used on an attribute on the class).